cmake/presets/ci.json

```
@@ -0,0 +1,575 @@
+{
+    "version": 4,
+    "configurePresets": [
+        {
+            "name": "base-ci",
+            "hidden": true,
+            "binaryDir": "${sourceDir}/build-${presetName}",
+            "generator": "Ninja",
+            "environment": {
+                "MPI_AS_ROOT": "OFF"
+            },
```

```suggestion
            },
            "condition": {
                "type": "equals",
                "lhs": "$env{GINKGO_PRESETS}",
                "rhs": "CI"
            },
```
We can use the environment to control the list here.
`GINKGO_PRESETS=CI cmake --list-presets configure ..` will show this additionally such that we can hide this long list to the users in the first side.